components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Engine/LegacyJobOrchestratorTests.cs

```diff
@@ -81,5 +84,85 @@ public class LegacyJobOrchestratorTests {
+                Assert.Equal("Succeded", publishNodesResult.First());
+            }
+
+            var tasks = new List<Task<JobProcessingInstructionModel>>();
+            for (var i = 0; i < 10; i++) {
+                tasks.Add(orchestrator.GetAvailableJobAsync(i.ToString(), new JobRequestModel()));
+            }
+
+            await Task.WhenAll(tasks);
+
+            Assert.Equal(2, tasks.Count(t => t.Result != null));
+            var distinctConfigurations = tasks
+                .Where(t => t.Result != null)
+                .Select(t => t.Result.Job.JobConfiguration)
+                .Distinct();
+            Assert.Equal(2, distinctConfigurations.Count());
+        }
+
+
+        [Theory]
+        [InlineData("Engine/publishednodes.json")]
+        [InlineData("Engine/publishednodeswithoptionalfields.json")]
+        [InlineData("Engine/pn_assets.json")]
+        [InlineData("Engine/pn_assets_with_optional_fields.json")]
+        public async Task DmApiUnPublishNodesonToEmptyConfiguration(string publishedNodesFile) {
+            var legacyCliModelProviderMock = new Mock<ILegacyCliModelProvider>();
+            var agentConfigProviderMock = new Mock<IAgentConfigProvider>();
+            var identityMock = new Mock<IIdentity>();
+            var newtonSoftJsonSerializer = new NewtonSoftJsonSerializer();
+            var jobSerializer = new PublisherJobSerializer(newtonSoftJsonSerializer);
+            var publishedNodesJobConverter = new PublishedNodesJobConverter(TraceLogger.Create(), newtonSoftJsonSerializer);
+
+            var legacyCliModel = new LegacyCliModel { PublishedNodesFile = publishedNodesFile, PublishedNodesSchemaFile = "Storage/publishednodesschema.json" };
+            legacyCliModelProviderMock.Setup(p => p.LegacyCliModel).Returns(legacyCliModel);
+            agentConfigProviderMock.Setup(p => p.Config).Returns(new AgentConfigModel());
+
+            var orchestrator = new LegacyJobOrchestrator(publishedNodesJobConverter, legacyCliModelProviderMock.Object, agentConfigProviderMock.Object, jobSerializer, TraceLogger.Create(), identityMock.Object);
+
+            using var payloads = new StreamReader(publishedNodesFile);
+            var unpublishNodesRequest = newtonSoftJsonSerializer.Deserialize<List<PublishedNodesEntryModel>>(await payloads.ReadToEndAsync());
+
+            foreach (var request in unpublishNodesRequest) {
+                var publishNodesResult = await orchestrator.UnpublishNodesAsync(request).ConfigureAwait(false);
+                Assert.Equal("Succeded", publishNodesResult.First());
+            }
+
+            var tasks = new List<Task<JobProcessingInstructionModel>>();
+            for (var i = 0; i < 10; i++) {
+                tasks.Add(orchestrator.GetAvailableJobAsync(i.ToString(), new JobRequestModel()));
+            }
+
+            await Task.WhenAll(tasks);
+
+            Assert.Equal(0, tasks.Count(t => t.Result != null));
```

Comment:
Not very important: I think FluentAssertion is better readable and shows more helpful debugging information as Assert.

Reply:
I agree.